src/main/java/uk/gov/pay/ledger/transaction/search/model/TransactionView.java

```diff
@@ -184,8 +184,11 @@ public static TransactionView from(Transaction transaction, int statusVersion) {
                     .withAgreementId(payment.getAgreementId())
                     .withDisputed(payment.getDisputed());
             if (payment.getState() != null) {
-                paymentBuilder = paymentBuilder
-                        .withState(ExternalTransactionState.from(payment.getState(), statusVersion));
+                if (payment.getAuthorisationMode() == AuthorisationMode.AGREEMENT && payment.getCanRetry() != null) {
```

Comment:
Is it possible to use one static method without having to check for authorisation_mode ?

`ExternalTransactionState.from(payment.getState(), statusVersion, payment.getCanRetry())`

for one-off can_retry field will always be null and will be excluded

Reply:
Yes, although we do check everywhere for authorisation mode before using canRetry